apps/blog/handlers/index.php

```diff
@@ -40,10 +40,12 @@
 		require_once ('apps/blog/lib/markdown.php');
 	}
 
-	foreach ($posts as $post) {
-		$post->url = '/blog/post/' . $post->id . '/' . URLify::filter ($post->title);
+	foreach ($posts as $_post) {
+		$post = $_post->orig();
+		$post->url = '/blog/post/' . $post->id . '/';
+		$post->fullurl = $post->url . URLify::filter ($post->title);
 		$post->tag_list = (strlen ($post->tags) > 0) ? explode (',', $post->tags) : array ();
-		$post->social_buttons = Appconf::blog ('Social Buttons');
+		$post->social_buttons = $appconf['Social Buttons'];
```

Comment:
I'd like to keep the `Appconf::*` style access to app settings. I'm thinking of eventually deprecating `$appconf` so the settings can be lazy-loaded and not automatically parsed on requests that don't refer to any settings (by removing this: https://github.com/jbroadway/elefant/blob/master/lib/Controller.php#L466-L467).